hooks/tk-multi-publish2/basic/nuke/publish_files.py

```diff
@@ -0,0 +1,182 @@
+                if node_file_path:
+                    if valid_paths[0] in node_file_path or valid_paths[1] in node_file_path:
+                        continue
+                    else:
+                        paths += "\n" + node_name + "  --->  " + node_file_path
+        if paths:
+            self.logger.error("Invalid paths! Try loading from shotgun.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Show invalid path(s)",
+                                      "text": "Invalid paths!\n{}".format(paths)
+                                  }
+                              }
+                              )
+            return False
+        return True
+
+    def _bbsize(self):
+        """
+        Checks for oversized bounding box for shotgun write nodes.
+
+        :return:True if all the write nodes have bounding boxes within limits
+        """
+        # Collecting sgtk write nodes
+        node = nuke.allNodes('WriteTank')
+        nodes = ""
+        for index, item in enumerate(node, 0):
+            bb = node[index].bbox()  # write node bbox
+            bb_height = bb.h()  # bbox height
+            bb_width = bb.w()  # bbox width
+
+            node_h = node[index].height()  # write node height
+            node_w = node[index].width()  # write node width
+            tolerance_h = (bb_height - node_h) / node_h * 100
+            tolerance_w = (bb_width - node_w) / node_w * 100
+
+            # Check if the size if over 5%(tolerance limit)
+            if tolerance_h > 5 or tolerance_w > 5:
+                nodes += "\n" + node[index]['name'].value()
+            if nodes:
+                self.logger.error(
+                    "Bounding Box resolution over the tolerance limit for write node(s).",
+                    extra={
+                        "action_show_more_info": {
+                            "label": "Show Info",
+                            "tooltip": "Show the write node(s)",
+                            "text": "Bounding Box resolution over the tolerance limit for:\n{}".format(nodes)
+                        }
+                    }
+                )
+                return False
+            return True
+
+    def validate(self, task_settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        status = True
+        status = self._bbsize() and status
+        status = self._file_paths() and status
```

Comment:
This is just checking file paths. Is there a way to check whether the input read nodes are pointing at PublishedFile entities? I think that is a valid check as well, though should just warn and not error out. Might be worth looking into the tk-multi-breakdown app to see how SG is checking this info.